IONFileViewerLib/IONFLVWDocumentOpener.swift

```diff
@@ -0,0 +1,68 @@
+import UIKit
+
+class IONFLVWDocumentOpener: NSObject, UIDocumentInteractionControllerDelegate {
+    
+    var documentInteractionController: UIDocumentInteractionController?
+    weak var viewController: UIViewController?
+    
+    init(viewController: UIViewController) {
+        self.viewController = viewController
+    }
+
+    func openDocumentFromUrl(url: URL, completion: @escaping (_ inner: IONFLVWCompletionHandler) -> Void) {
+        IONFLVWFileDownloader.downloadFile(url: url, completion: { (_, inner: IONFLVWURLCompletionHandler) in
```

Comment:
Could we possibly use weak self here to avoid potential memory leaks? And then we can safely unwrap self. So something like this following:

```
IONFLVWFileDownloader.downloadFile(url: url) { [weak self] (_, inner: IONFLVWURLCompletionHandler) in
        guard let self = self else { return }
        
        do {
                ...
        }

```

Reply:
Ok yeah that makes sense!